frontend/src/features/admin-form/responses/FeedbackPage/FeedbackTable.tsx

```diff
@@ -0,0 +1,149 @@
+import React, { useEffect } from 'react'
+import { Column, usePagination, useSortBy, useTable } from 'react-table'
+import { Box, Icon, Table, Tbody, Td, Th, Thead, Tr } from '@chakra-ui/react'
+
+import { ProcessedFeedbackMeta } from '~shared/types/form'
+
+import { BxsChevronDown } from '~assets/icons/BxsChevronDown'
+import { BxsChevronUp } from '~assets/icons/BxsChevronUp'
+
+type Data = {
+  index: number
+  date: string
+  feedback: string
+  rating: number
+}
+
+export const FeedbackTable = ({
+  feedbackData,
+  currentPage,
+}: {
+  feedbackData: ProcessedFeedbackMeta[] | undefined
+  currentPage: number
+}) => {
+  const columns = React.useMemo<Column<Data>[]>(
+    () => [
+      {
+        Header: '#',
+        accessor: 'index',
+        sortType: 'basic',
+        width: '5rem',
+      },
+      {
+        Header: 'Date',
+        accessor: 'date',
+        sortType: (rowA, rowB) => {
+          const dateA = new Date(rowA.values.id)
+          const dateB = new Date(rowB.values.id)
+          return dateA > dateB ? 1 : -1
+        },
+        width: '10rem',
+      },
+      {
+        Header: 'Feedback',
+        accessor: 'feedback',
+        sortType: 'basic',
+        width: undefined,
+      },
+      {
+        Header: 'Rating',
+        accessor: 'rating',
+        sortType: 'basic',
+        width: '12rem',
+      },
+    ],
+    [],
+  )
+
+  const data = React.useMemo(() => {
+    return feedbackData
+      ? feedbackData.map((feedback) => {
+          return {
+            index: feedback.index,
+            date: feedback.date,
+            feedback: feedback.comment,
+            rating: feedback.rating,
+          }
+        })
+      : []
+  }, [feedbackData])
+
+  const { prepareRow, headerGroups, page, gotoPage } = useTable<Data>(
+    {
+      columns,
+      data,
+      initialState: { pageIndex: currentPage, pageSize: 9 },
+    },
+    useSortBy,
```

Comment:
nit: how would sort work? Would it give the users the perception that it's sorting on the entire feedback instead of just a single page? Maybe it's a question to ask @pearlyong hehe

Reply:
Actually it does sort on the entire feedback
e.g. if I am now on page 3 and I click sort by rating, then I will see the 19th - 27th entries based on the new sorted order
@pearlyong to confirm this is desired behavior